common/pinot/GenericClient_mock.go

```diff
@@ -0,0 +1,100 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Code generated by MockGen. DO NOT EDIT.
+// Source: interfaces.go
```

Comment:
how did you generate this? I don't see common/pinot/interfaces.go in the diff

Reply:
https://github.com/uber/cadence/blob/master/common/pinot/interfaces.go
I implemented it previously. Since ESVisibilityStore were using a generic es client, I used the same pattern.

Reply:
I meant interface.go doesn't have any change in the PR. We usually generate mocks by adding go:generate lines to the interface file. example: https://github.com/uber/cadence/blob/68706a6219d11a2113d9bc1d257997c85da0f433/client/admin/interface.go#L31

Reply:
This is because of a go version issue. The previous branch was massed up, so I just copied and pasted the GenericClient_mock.go. I've added the comment back.